src/devices/nodon.ts

```diff
@@ -38,33 +63,44 @@ const definitions: Definition[] = [
             await reporting.currentPositionLiftPercentage(endpoint);
             await reporting.currentPositionTiltPercentage(endpoint);
         },
-        exposes: [e.cover_position()],
+        exposes: [e.cover_position_tilt()],
         ota: ota.zigbeeOTA,
     },
     {
         zigbeeModel: ['SIN-4-1-20'],
         model: 'SIN-4-1-20',
         vendor: 'NodOn',
         description: 'Multifunction relay switch',
-        extend: extend.switch(),
+        fromZigbee: [fz.identify, fz.on_off, fz.command_toggle, fz.command_on, fz.command_off, fz.power_on_behavior, fzLocal.impulse_time],
+        toZigbee: [tz.on_off, tz.power_on_behavior, tzLocal.impulse_time],
+        exposes: [
+            e.switch(),
+            e.action(['identify', 'on', 'off', 'toggle']),
+            e.power_on_behavior(),
+        ],
+        ota: ota.zigbeeOTA,
         configure: async (device, coordinatorEndpoint, logger) => {
             const endpoint = device.getEndpoint(1);
             await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff']);
             await reporting.onOff(endpoint);
-            // NodOn Attribute
-            await endpoint.read('genOnOff', ['nodonTransitionTime']);
         },
         endpoint: (device) => {
             return {default: 1};
         },
-        ota: ota.zigbeeOTA,
     },
     {
-        zigbeeModel: ['SIN-4-1-20_PRO'],
         model: 'SIN-4-1-20_PRO',
+        zigbeeModel: ['SIN-4-1-20_PRO'],
         vendor: 'NodOn',
         description: 'Multifunction relay switch',
-        extend: extend.switch(),
+        fromZigbee: [fz.identify, fz.on_off, fz.command_toggle, fz.command_on, fz.command_off, fz.power_on_behavior, fzLocal.impulse_time],
+        toZigbee: [tz.on_off, tz.power_on_behavior, tzLocal.impulse_time],
```

Comment:
I see nothing is added to expose for the `impulse_time`, instead of adding a tz/fz and expose, could you use the modernExtend instead? Example https://github.com/Koenkk/zigbee-herdsman-converters/blob/4ce6450b2a55761f7e9f38688b645748885b876d/src/devices/xiaomi.ts#L987